src/Harald.WebApi/Features/Connections/Infrastructure/Persistence/FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandler.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Harald.WebApi.Domain;
+using Harald.WebApi.Domain.Queries;
+using Harald.WebApi.Features.Connections.Domain.Model;
+using Harald.WebApi.Features.Connections.Domain.Queries;
+
+namespace Harald.WebApi.Features.Connections.Infrastructure.Persistence
+{
+    public class FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandler : IQueryHandler<
+        FindConnectionsBySenderTypeSenderIdChannelTypeChannelId, IEnumerable<Connection>>
+    {
+        private readonly ICapabilityRepository _capabilityRepository;
+
+        public FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandler(
+            ICapabilityRepository capabilityRepository)
+        {
+            _capabilityRepository = capabilityRepository;
+        }
+
+        public async Task<IEnumerable<Connection>> HandleAsync(
+            FindConnectionsBySenderTypeSenderIdChannelTypeChannelId query)
+        {
+         
+            
+            
+            if (query.ChannelType != null && query.ChannelType.GetType() != typeof(ChannelTypeSlack) )
+            {
+                return Enumerable.Empty<Connection>();
+            }
+
+            if (query.SenderType != null && query.SenderType.GetType() != typeof(SenderTypeCapability))
+            {
+                return Enumerable.Empty<Connection>();
+            }
+
+            var capabilities = await _capabilityRepository.GetAll();
+
+            if (query.ChannelId != null)
+            {
+                capabilities = capabilities
+                    .Where(c =>
+                        c.SlackChannelId.Equals(query.ChannelId)
+                    );
+            }
+
+            if (query.SenderId != null)
+            {
+                Guid capabilityId;
+                var capabilityIdIsValid = Guid.TryParse(query.SenderId, out capabilityId);
```

Comment:
Consider using the TryParse operation directly in the if statement and have a look at out-variables @ https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#out-variables which are available for the rest of the method scope (not just the nested if)

Reply:
The ultimate location of the conversion would be in a CapabilityId value object what could encapsulate the rule of capability ids having to be a Guid. that said second best is to somehow indicate my intention of validating the capability id based on it having to be a Guid. Hence the very explicit capabilityIdIsValid.

I tend to miss variable definitions written in the out parameter of methods so I declare the vessel or variable before the method containing the out parameter in this case TryParse.